comments/comments.go

```diff
@@ -1,33 +1,32 @@
-package main
+package comments
 
 import (
-	"errors"
+	"bytes"
 	"fmt"
 	"go/ast"
+	"go/format"
 	"go/parser"
 	"go/printer"
 	"go/token"
-	"log"
-	"os"
+	"io/ioutil"
 	"sort"
 	"strings"
 )
 
-func main() {
-	fset := token.NewFileSet()
-	if len(os.Args) == 0 {
-		log.Fatal(errors.New("args is too short"))
-	}
-	fname := os.Args[1]
-	_, err := os.Stat(fname)
-	if err != nil {
-		log.Fatal(errors.New("file not exists"))
+//Process is adding comment templates to should-commented variable names, structure definitions, and function names
+func Process(filename string, src []byte) ([]byte, error) {
+	if src == nil {
+		b, err := ioutil.ReadFile(filename)
```

Comment:
The function should do just one thing.
`Process` should just add comment templates, but should not access files.

I recommend that just panic in this case.

```go
if src == nil {
    panic ("src is nil")
}
```